tests/Test.ADAL.NET.Integration/SovereignCloudsTests.cs

```diff
@@ -60,16 +96,78 @@ public void Initialize()
+        public async Task AuthorityNotInInstanceCache_InstanceDiscoverCallMadeTestAsync()
+        {
+            string content = @"{
+                            ""tenant_discovery_endpoint"":""https://login.microsoftonline.com/tenant/.well-known/openid-configuration"",
+                            ""api-version"":""1.1"",
+                            ""metadata"":[{
+                                ""preferred_network"":""login.microsoftonline.com"",
+                                ""preferred_cache"":""login.windows.net"",
+                                ""aliases"":[
+                                    ""login.microsoftonline.com"",
+                                    ""login.windows.net"",
+                                    ""login.microsoft.com"",
+                                    ""sts.windows.net""]}]}";
+
+            // Assure instance cache is empty
+            Assert.AreEqual(0, InstanceDiscovery.InstanceCache.Count());
+
+            // create AuthenticationContext with common authority
+            var authenticationContext =
+                new AuthenticationContext(TestConstants.DefaultAuthorityCommonTenant, false, new TokenCache());
+
+            MockHelpers.ConfigureMockWebUI(new AuthorizationResult(AuthorizationStatus.Success,
+                           TestConstants.DefaultRedirectUri + "?code=some-code"));
+            HttpMessageHandlerFactory.AddMockHandler(new MockHttpMessageHandler(TestConstants.GetTokenEndpoint(TestConstants.DefaultAuthorityCommonTenant))
+            {
+                Method = HttpMethod.Post,
+                ResponseMessage = MockHelpers.CreateSuccessTokenResponseMessage()
+            });
+
+            HttpMessageHandlerFactory.AddMockHandler(MockHelpers.CreateInstanceDiscoveryMockHandler(TestConstants.GetDiscoveryEndpoint(TestConstants.DefaultAuthorityBlackforestTenant), content));
+
+            AuthenticationResult authenticationResult = await authenticationContext.AcquireTokenAsync(TestConstants.DefaultResource,
+                TestConstants.DefaultClientId,
+                TestConstants.DefaultRedirectUri, platformParameters);
+
+            Assert.AreEqual(1, InstanceDiscovery.InstanceCache.Count());
+            Assert.AreEqual(true, InstanceDiscovery.InstanceCache.Keys.Contains("login.microsoftonline.com"));
+            // DE cloud authority not yet included in instance cache
+            Assert.AreEqual(false, InstanceDiscovery.InstanceCache.Keys.Contains("login.microsoftonline.de"));
+
+            // One mock remaining for instance discovery call
+            Assert.AreEqual(1, HttpMessageHandlerFactory.MockHandlersCount());
+            
+            // Instance discovery makes another call because DE cloud authority is not in the cache
+            var entry = await InstanceDiscovery.GetMetadataEntry(new Uri("https://login.microsoftonline.de"), true, new CallState(Guid.NewGuid())).ConfigureAwait(false);
```

Comment:
why this your test case making this call? It should be ADAL code calling the InstanceDiscovery not the test itself. remove this GetMetadataEntry() call.

Reply:
I fixed this